requirements.lock

```diff
@@ -10,5 +10,5 @@
 -e file:.
 colorama==0.4.6
     # via mepo
-pyyaml==6.0.1
+pyyaml==6.0.2
```

Comment:
This is interesting. Spack doesn't seem to see the version of pyyaml! I'll ask on the slack...

Reply:
We "shouldn't" need to version control the lock files. They are to be generated by `rye` during `rye sync`. I removed them, but they came back while merging in the changes from `main`.

Reply:
@mathomp4 Removed dependencies on `requirements**.lock`. This should be good to go.